### ISSUE TITLE
Federated sharing activity gets triggered twice.

### ISSUE THREAD

**Issue by schiessle** [nextcloud/server#17022]
I just tested it with Nextcloud 16 and can reproduce it reliable:

1. user1@server1 shares a file to user2@server2
2. user2@server2 accept the share
3. look at the activities of user1@server1

user1 sees the activity that user2@server2 accepted the federated share twice:

![image](https://user-images.githubusercontent.com/1589737/64351965-8369ca00-cffb-11e9-80aa-e2c8b01fd4f0.png)

Expected behavior: Only one notification shows up

I can think of two different places where this happens:

1. activity creation (but I think that's less likely)
2. server2 sends for whatever reasons twice the "share accepted" message to server1

**Comment by gary-kim** [comment]
Confirmed the issue exists:
![Screenshot from 2019-09-07 17-34-43](https://user-images.githubusercontent.com/47195730/64472761-e2634680-d195-11e9-8d23-8a84459d9f45.png)

**Comment by szaimen** [comment]
I suppose this issue is still valid? If not, please close this issue!

**Comment by MrRinkana** [comment]
This happened to me today using nc 24.0.4, but only once. I shared a folder to a friend on another server and he got two popups "accept share" which he accepted both. Since then he removed the share, I re-shared and he only got one. He also shared a folder with me and I also only got one notification.

Maybe only happens on the first share to an external nc server? Before the servers "know" each other? I'm just guessing.

I haven't found anything in the logs (accessible from the web UI), but I did get two emails from my server confirming he accepted the share twice.
